packages/currency/src/native.ts

```diff
@@ -1,7 +1,11 @@
-import { RequestLogicTypes } from '@requestnetwork/types';
-import { NativeCurrency, NativeCurrencyType } from './types';
+import { CurrencyTypes, RequestLogicTypes } from '@requestnetwork/types';
+import { NamedNativeCurrency } from './types';
 
-export const nativeCurrencies: Record<NativeCurrencyType, (NativeCurrency & { name: string })[]> = {
+type NativeEthCurrency = NamedNativeCurrency & { network: CurrencyTypes.EvmChainName };
```

Comment:
1/ Generic question: why having a different type per blockchain type?

2/
Watch out: 
```
    {
      symbol: 'NEAR',
      decimals: 24,
      name: 'Near',
      network: 'aurora',
    },
```
Was defined within `RequestLogicTypes.CURRENCY.ETH`, as it seemed complex to define one Currency type per chain. These new types can enforce the confusion, maybe it's easier to make the difference at this stage with `NativeNearCurrency` (but cf. 1/).

Option 1/n: `NativeAMCurrency`, AM standing for `AccountModel`, opposed to UTXO (https://www.horizen.io/blockchain-academy/technology/expert/utxo-vs-account-model/#:~:text=The%20UTXO%20model%20is%20a,constructions%2C%20as%20well%20a%20sharding.)
Option 2/n: `NativeVMCurrency`: Common type for VM chains native tokens
...

3/ Consider usign `Evm` instead of `Eth`

Reply:
I fixed it in https://github.com/RequestNetwork/requestNetwork/pull/1056/commits/d6ae451ecd548c5f27b6686fc721891e7eb9d2cc with the following bit of code:
```ts
type NativeEthCurrency = NamedNativeCurrency & {
  network: CurrencyTypes.EvmChainName | CurrencyTypes.NearChainName;
};
```

> why having a different type per blockchain type

The goal was to have a different type per currency type.

IMO, if we can specify the type, let's do it; it brings clarity to the model. The change already revealed small flaws in the current design (which are OK flaws):
- **flaw 1**: because chain names are not unique across different chain types, a collision is possible for currency types that span across multiple chain types, like `RequestLogicTypes.CURRENCY.ETH` that spans across EVM-type and NEAR-type chains
- **flaw 2**: the currency type `RequestLogicTypes.CURRENCY.ETH`'s name is a bit confusing

I think that's great: it made it more obvious. I didn't want to change any logic in this PR, so I haven't refactored anything related to currencies. Let's keep it in mind tho! I didn't know about "UTXO vs. Account Model", thanks!